crates/matrix-sdk/src/notification_settings/mod.rs

```diff
@@ -124,7 +126,43 @@ impl NotificationSettings {
         Ok(())
     }
 
-    /// Sets the notification mode for a room.
+    /// Set the default notification mode for a room kind.
+    ///
+    /// # Arguments
+    ///
+    /// * `encrypted` - `true` if the mode is for encrypted rooms
+    /// * `one_to_one` - `true` if the mode if for direct chats
+    /// * `mode` - the new default mode
+    pub async fn set_default_room_notification_mode(
+        &self,
+        encrypted: bool,
+        one_to_one: bool,
```

Comment:
Having two boolean parameters following each other in a function signature makes it harder for users to know which corresponds to what, and it makes it way too easy to mistake both, shooting oneself in the feet. A slightly better but more verbose API would consist in having two enums, `enum Encrypted { Yes, No }; enum IsDirectMessage { Yes, No }` and use these in the signature here, to make it impossible to be confused about the role of which boolean parameter, when calling the function.

(aka, roses are red, violets are blue, enums are great, booleans are poo)

Reply:
done.